river/command.zig

```diff
@@ -117,3 +117,59 @@ pub fn errToMsg(err: Error) [:0]const u8 {
         Error.Other => unreachable,
     };
 }
+
+/// Parses a list of options specified by T
+/// args[0] must be the first option if any
+/// offset will be set to the index of the first entry in args that is not an option
+/// See test "optinal args" for an example of T
+pub fn parse(args: []const []const u8, offset: *usize, comptime T: type) T {
+    // The empty {} make sure there is a default value for all members
+    var ret: T = T{};
+    // Make sure the format of T is valid
+    if (@typeInfo(T) != .Struct) @compileError("T must be a struct");
+    inline for (@typeInfo(T).Struct.fields) |field| {
+        if (@typeInfo(field.field_type) != .Bool)
+            @compileError("unsupported option type " ++ field.field_type);
+        if (field.default_value.?)
+            @compileError("bool options must be false by default");
+        if (field.name[0] != '-')
+            // TODO Should we really have this check? If yes test ascii.isAlpha(field.name[1])
+            // Figure out how long arguments should be is -a a valid argument
+            @compileError("option names must start with a single - followed by an alphabetical cahr");
+    }
+
+    // Counts the amout of parsed args
+    var i: usize = 0;
+    // Do the actual parsing of args
+    for (args) |arg| {
+        var found = false;
+        inline for (@typeInfo(T).Struct.fields) |field| {
+            if (std.mem.eql(u8, arg, field.name)) {
+                found = true;
+                @field(ret, field.name) = true;
+            }
+        }
+        // stop parsing options when no options are left
+        if (!found) break;
+        i += 1;
+    }
+
+    offset.* = i;
```

Comment:
This loop can be written a little more cleanly:
```suggestion
    offset.* = 0;
    for (args) |arg| {
        inline for (@typeInfo(T).Struct.fields) |field| {
            if (std.mem.eql(u8, arg, field.name)) {
                @field(ret, field.name) = true;
                offset.* += 1;
                continue;
            }
        }
        // There are no options left
        break;
    }
```